spec/grammar-spec.coffee

```diff
@@ -113,37 +113,359 @@ describe "SQL grammar", ->
     expect(tokens[1]).toEqual value: 'Test', scopes: ['source.sql', 'string.quoted.double.sql']
     expect(tokens[2]).toEqual value: '"', scopes: ['source.sql', 'string.quoted.double.sql', 'punctuation.definition.string.end.sql']
 
-  it 'tokenizes the time type', ->
-    {tokens} = grammar.tokenizeLine('TIME')
-    expect(tokens[0]).toEqual value: 'TIME', scopes: ['source.sql', 'storage.type.sql']
-
-    {tokens} = grammar.tokenizeLine('TIME WITH TIME ZONE')
-    expect(tokens[0]).toEqual value: 'TIME', scopes: ['source.sql', 'storage.type.sql']
-    expect(tokens[2]).toEqual value: 'WITH TIME ZONE', scopes: ['source.sql', 'storage.type.sql']
-
-    {tokens} = grammar.tokenizeLine('TIME(1)WITHOUT TIME ZONE\'23:00\'')
-    expect(tokens[0]).toEqual value: 'TIME', scopes: ['source.sql', 'storage.type.sql']
-    expect(tokens[2]).toEqual value: '1', scopes: ['source.sql', 'constant.numeric.sql']
-    expect(tokens[4]).toEqual value: 'WITHOUT TIME ZONE', scopes: ['source.sql', 'storage.type.sql']
-
-  it 'tokenizes the timestamp type', ->
-    {tokens} = grammar.tokenizeLine('TIMESTAMP ( 12 )  WITH TIME ZONE')
-    expect(tokens[0]).toEqual value: 'TIMESTAMP', scopes: ['source.sql', 'storage.type.sql']
-    expect(tokens[2]).toEqual value: '12', scopes: ['source.sql', 'constant.numeric.sql']
-    expect(tokens[4]).toEqual value: 'WITH TIME ZONE', scopes: ['source.sql', 'storage.type.sql']
-
-  it 'tokenizes the timestamptz type', ->
-    {tokens} = grammar.tokenizeLine('timestamptz')
-    expect(tokens[0]).toEqual value: 'timestamptz', scopes: ['source.sql', 'storage.type.sql']
-
-    {tokens} = grammar.tokenizeLine('TIMESTAMPTZ(2)NOT NULL')
-    expect(tokens[0]).toEqual value: 'TIMESTAMPTZ', scopes: ['source.sql', 'storage.type.sql']
-    expect(tokens[2]).toEqual value: '2', scopes: ['source.sql', 'constant.numeric.sql']
-
-  it 'tokenizes the timetz type', ->
-    {tokens} = grammar.tokenizeLine('timetz (2)')
-    expect(tokens[0]).toEqual value: 'timetz', scopes: ['source.sql', 'storage.type.sql']
-    expect(tokens[2]).toEqual value: '2', scopes: ['source.sql', 'constant.numeric.sql']
+  it 'tokenizes column types', ->
```

Comment:
This is a rather massive test.  We don't have to test for everything, just smoke-test 1-2 from each category (and consider separating into different tests).